allennlp/data/dataset_readers/dataset_utils/ontonotes.py

```diff
@@ -0,0 +1,418 @@
+                    if segment[-1] == ")":
+                        # The span begins and ends at this word (single word span).
+                        cluster_id = int(segment[1:-1])
+                        clusters[cluster_id].append((word_index, word_index))
+                    else:
+                        # The span is starting, so we record the index of the word.
+                        cluster_id = int(segment[1:])
+                        coref_stacks[cluster_id].append(word_index)
+                else:
+                    # The span for this id is ending, but didn't start at this word.
+                    # Retrieve the start index from the document state and
+                    # add the span to the clusters for this id.
+                    cluster_id = int(segment[:-1])
+                    start = coref_stacks[cluster_id].pop()
+                    clusters[cluster_id].append((start, word_index))
+
+    @staticmethod
+    def _process_span_annotations_for_word(annotations: List[str],
+                                           span_labels: List[List[str]],
+                                           current_span_labels: List[Optional[str]]):
+        """
+        Given a sequence of different label types for a single word and the same labels
+        for the previous word, compute the BIO tag for each label and append to a list.
+        annotations: ``List[str]``
+            A list of labels to compute BIO tags for.
+        span_labels : ``List[List[str]]``
+            A list of lists, one for each annotation, to incrementally collect
+            the BIO tags for a sequence.
+        current_span_labels : ``List[Optional[str]]``
+            The currently open span per annotation type, or ``None`` if there is no open span.
+        """
+        for annotation_index, annotation in enumerate(annotations):
+            label = annotation.strip("()*")
+
+            if "(" in annotation:
+                # Entering into a span for a particular semantic role label.
+                # We append the label and set the current span for this annotation.
+                bio_label = "B-" + label
+                span_labels[annotation_index].append(bio_label)
+                current_span_labels[annotation_index] = label
+
+            elif current_span_labels[annotation_index] is not None:
+                # If there's no '(' token, but the current_span_label is not None,
+                # then we are inside a span.
+                bio_label = "I-" + current_span_labels[annotation_index]
+                span_labels[annotation_index].append(bio_label)
+            else:
+                # We're outside a span.
+                span_labels[annotation_index].append("O")
+
+            # Exiting a span, so we reset the current span label for this annotation.
+            if ")" in annotation:
+                current_span_labels[annotation_index] = None
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'Ontonotes':
```

Comment:
I don't see a purpose for this method - if you're going to use Ontonotes, you probably know you're going to use Ontonotes, and won't be building this `from_params`.